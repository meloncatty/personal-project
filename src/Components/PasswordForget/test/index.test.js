import React from 'react'
import { PasswordForget } from '../'
import { shallow } from 'enzyme'

describe('PasswordForget', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<PasswordForget />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have default state', () => {
    const wrapper = shallow(<PasswordForget />)
    const expected = {
      email: '',
      submitSuccessful: false,
      error: false
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('should update state on submit', () => {
    const wrapper = shallow(<PasswordForget />)
    const mockEvent = {preventDefault: () => {}}
    const expected = {
      email: '',
      submitSuccessful: false,
      error: false
    }
    wrapper.find('form').simulate('submit', mockEvent)

    expect(wrapper.state()).toEqual(expected)
  })
})
