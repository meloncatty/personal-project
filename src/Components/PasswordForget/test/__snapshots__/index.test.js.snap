// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForget should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordForget />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="reset-password"
        onSubmit={[Function]}
>
        <label>
                Reset Your Password
        </label>
        <input
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
        />
        <button
                disabled={true}
                type="submit"
        >
                Reset Password
        </button>
</form>,
      "className": "form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Reset Your Password
</label>,
          <input
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Reset Password
</button>,
          false,
          false,
        ],
        "className": "reset-password",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Your Password",
          },
          "ref": null,
          "rendered": "Reset Your Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Password",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Reset Password",
          "type": "button",
        },
        false,
        false,
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="reset-password"
          onSubmit={[Function]}
>
          <label>
                    Reset Your Password
          </label>
          <input
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Reset Password
          </button>
</form>,
        "className": "form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Reset Your Password
</label>,
            <input
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Reset Password
</button>,
            false,
            false,
          ],
          "className": "reset-password",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset Your Password",
            },
            "ref": null,
            "rendered": "Reset Your Password",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset Password",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Reset Password",
            "type": "button",
          },
          false,
          false,
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
